src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,41 @@
+import ModalCloseIcon from "@/icons/ModalCloseIcon";
+import { useEffect } from "react";
+
+type ModalProps = {
+  isModalOpen: boolean;
+  setIsModalOpen: (isOpen: boolean) => void;
+  children: React.ReactNode;
+};
+
+const Modal = ({ children, isModalOpen, setIsModalOpen }: ModalProps) => {
+  useEffect(() => {
```

Comment:
When the modal is open, you might want to prevent the main page from scrolling.
`
useEffect(() => {
  if (isModalOpen) {
    document.body.style.overflow = 'hidden';
    return () => {
      document.body.style.overflow = 'unset';
    };
  }
}, [isModalOpen]);
`